src/preamble/02_what_is_a_module_and_a_package.md

```diff
@@ -1 +1,194 @@
 # What is a module and a package?
+
+## TL;DR
+
+A **module** is a unit of code defined by GHC and combined by Cabal to create
+packages. A **package** is either a library (a re-usable collection of modules),
```

Comment:
I'm not versed in cabal terminology, but isn't what you call a 'package` normally called a `component`? And a package is a set of components, all of which are defined in the .cabal file for the package.

Here is an excerpt from #hackage that mentions some of the concepts, with links to the technical side of them (it also mentions a 'unit' --- I wonder what it is and I'd love to learn it from this chapter, if it's in scope):

```
16:49 < mikolaj> danidiaz: to muddle the issues further, here is a lengthy discussion about "the relationship between ComponentId, UnitId, MungedPackageId, PackageId" https://github.com/haskell/cabal/issues/5809
16:51 < mikolaj> if you could cross-check what's there, what's in the new code, what's already documented, etc., and report in the issue that would be helpful; bonus points if any actual docs improvements follow
17:19 < hsyl20> mikolaj danidiaz I've spent quite some time understanding and writing some notes on the GHC side: https://github.com/ghc/ghc/blob/ce1b8f4208530fe6449506ba22e3a05048f81564/compiler/GHC/Unit.hs#L25
17:19 < hsyl20> (github link while gitlab is down...)
```

Reply:
I was about to comment about this too. I've tried to summarize the terminology here: https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/compiler/units#background

It would be great if we could use the same everywhere :)

Reply:
Good catch!